gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,383 @@
+                    handlerSpec: handlerSpec,
+                    isError: true,
+                    message: "inner promise error",
+                    err: err
+                });
+            });
+        }
+        catch (err) {
+            promiseTogo.resolve({
+                solutionId: solutionID,
+                handlerSpec: handlerSpec,
+                isError: true,
+                message: "outer try error",
+                err: err
+            });
+        }
+        return promiseTogo;
+    };
+
+    var solutionsToFetch = solutions;
+    if (options.solutionsList) {
+        solutionsToFetch = {};
+        fluid.each(solutions, function (solution, solutionID) {
+            if (options.solutionsList.indexOf(solutionID) >= 0) {
+                solutionsToFetch[solutionID] = solution;
+            }
+        });
+    }
+
+    fluid.each(solutionsToFetch, function (solution, solutionID) {
+        var newSolution = gpii.lifecycleManager.transformSolutionSettings(solution, gpii.settingsHandlers.changesToSettings);
+
+        if (solutionID === "com.freedomscientific.jaws") {
+            newSolution = gpii.flowManager.jawsVoiceProfileAdjustment(newSolution, localResolver);
+        }
+
+        fluid.each(newSolution.settingsHandlers, function (handlerSpec) {
+            //if applicationSettings is unspecified, dont attempt to get settings
+            if (!handlerSpec.supportedSettings) {
+                return;
+            }
+            handlerSpec.settings = handlerSpec.supportedSettings;
+            // TODO GPII-228 This is temporary debugging code while we work through
+            // the processing that is missing from this chain that transforms the paths
+            // in capabilitiesTransforms to the schema information needed for the settings
+            // handler
+            if (handlerSpec.type === "gpii.windows.spiSettingsHandler") {
+                fluid.each(handlerSpec.settings, function (handler, handlerKey) {
+                    if (handlerSpec.capabilitiesTransformations[handlerKey].path.transform) {
+                        handlerSpec.settings[handlerKey].path =
+                            handlerSpec.capabilitiesTransformations[handlerKey].path.transform.input;
+                    };
+                    if (handlerSpec.capabilitiesTransformations[handlerKey].path.literalValue) {
+                        handlerSpec.settings[handlerKey].path =
+                            handlerSpec.capabilitiesTransformations[handlerKey].path.literalValue;
+                    };
+                });
+            }
+            handlerSpec = resolver.resolve(handlerSpec, {});
+            handlerGetPromises.push(safeHandlerGet(invokeSettingsHandlerGet, solutionID, handlerSpec));
+        });
+    });
+    var promisesSequence = fluid.promise.sequence(handlerGetPromises);
+    promisesSequence.then(function (responses) {
+
+        var finalPayload = gpii.flowManager.formatRawCapturedSettings(responses);
```

Comment:
The formatting of read settings should be moved out of the separate function I mentioned in a previous comment. The read() function just returns the result from invokeSettingsHandlerGet(). It is more flexible to leave the formatting to the caller function per case basis.

Reply:
This has been moved in to the transforming promise chain.